app/dashboard/management/commands/sync_geth.py

```diff
@@ -15,117 +15,47 @@
+from dashboard.helpers import UnsupportedSchemaException
+from dashboard.utils import BountyNotFoundException, get_bounty, process_bounty, start_ipfs
 
-    return realtime_callback if realtime else faux_realtime_callback
+warnings.filterwarnings("ignore", category=DeprecationWarning) 
 
 
 class Command(BaseCommand):
 
     help = 'syncs bounties with geth'
 
     def add_arguments(self, parser):
-        parser.add_argument(
-            '--network',
-            dest='network',
-            type=str,
-            default=settings.DEFAULT_NETWORK,
-            help="network (optional)",
-        )
-
-        parser.add_argument(
-            '--provider',
-            dest='provider',
-            type=str,
-            default='default',
-            help="provider (optional)",
-        )
+        parser.add_argument('network')
 
     def handle(self, *args, **options):
 
+        # config
+        network = options['network']
+
         # setup
-        bounty_contract_address, infura_host, custom_geth_details, is_testnet = get_network_details(options['network'])
-        print("****************************************")
-        print("connecting {} {} ....".format(options['network'], options['provider']))
-        print("****************************************")
-
-        start_time = int(timezone.now().strftime("%S"))
-        web3 = getWeb3(options['network'], options['provider'])
-        end_time = int(timezone.now().strftime("%S"))
-        connect_time = end_time - start_time
-
-        print("****************************************")
-        print("connected to {} (took {} s)".format(web3.currentProvider.__class__, connect_time))
-        print("****************************************")
-
-        # get past event topics
-        try:
-            print('- attempting realtime')
-            callback = get_callback(web3, bounty_contract_address, True)
-            web3.eth.filter('pending').watch(callback)
-            print('- sleeping')
-            while True:
-                time.sleep(1)
-        except requests.exceptions.HTTPError:
-            print("- realtime not working. attempting to faux realtime! #YOLO")
-            callback = get_callback(web3, bounty_contract_address, False)
-            last_blockNumber = 0
-            while True:
-                blockNumber = web3.eth.blockNumber
-                if blockNumber != last_blockNumber:
-                    print("-- new block: {}".format(blockNumber))
-                    callback(blockNumber)
-                last_blockNumber = blockNumber
-                time.sleep(POLL_SLEEP_TIME)
+        start_ipfs()
+
+        # iterate through all the bounties
+        bounty_enum = 0
+        more_bounties = True
+        while more_bounties:
+            try:
+
+                # pull and process each bounty
+                bounty = get_bounty(bounty_enum, network)
+                print(f"Processing bounty {bounty_enum}")
+                process_bounty(bounty)
```

Comment:
Maybe break this out into an `else` statement.  Which statement are you concerned will fail?  Normally I put that one statement in the `try`, and then the `else` is the code that will run if the code under `try` works.

Reply:
i dont think i can do this with this one... the exception handler is meant for the entire block